src/sources/docker_logs.rs

```diff
@@ -75,26 +77,39 @@ impl Default for DockerLogsConfig {
 }
 
 impl DockerLogsConfig {
-    fn container_name_included<'a>(
+    fn container_name_or_id_included<'a>(
         &self,
         id: &str,
         names: impl IntoIterator<Item = &'a str>,
     ) -> bool {
-        if let Some(include_containers) = &self.include_containers {
-            let id_flag = include_containers
-                .iter()
-                .any(|include| id.starts_with(include));
+        let containers: Vec<String> = names.into_iter().map(|i| i.into()).collect();
 
-            let name_flag = names.into_iter().any(|name| {
-                include_containers
-                    .iter()
-                    .any(|include| name.starts_with(include))
-            });
+        // Containers are included by default
+        let mut included: bool = true;
 
-            id_flag || name_flag
-        } else {
-            true
+        // Container remains included if include_containers is specified
+        // and container name or ID matches the inclusion list
+        if let Some(include_list) = &self.include_containers {
+            included = Self::name_or_id_matches(id, &containers, include_list);
+        }
+
+        // Container gets excluded if exclude_containers is specified
+        // and container name or ID matches the exclusion list
+        if let Some(exclude_list) = &self.exclude_containers {
+            included = !(Self::name_or_id_matches(id, &containers, exclude_list));
         }
+
+        included
+    }
+
+    fn name_or_id_matches(id: &str, names: &Vec<String>, items: &Vec<String>) -> bool {
+        let id_matches = items.iter().any(|flag| id.starts_with(flag));
+
+        let name_matches = names
+            .into_iter()
+            .any(|name| items.iter().any(|item| name.starts_with(item)));
+
+        id_matches || name_matches
```

Comment:
Note that calculating them separately like this and then oring them together means `name_matches` may be calculated even if `id_matches` is true and the `||` would short-circuit. ("may be" because the compiler may determine it can do this itself and do the optimization anyways.) If this is in a hot path, that could lead to reduced performance. Compare:

```suggestion
        items.iter().any(|flag| id.starts_with(flag))
            || names
                .into_iter()
                .any(|name| items.iter().any(|item| name.starts_with(item)))
```

Reply:
Great suggestion, and good to keep in mind re: potentially unnecessary calls